src/core/dbus-manager.c

```diff
@@ -2937,6 +2937,146 @@ static int method_dump_unit_descriptor_store(sd_bus_message *message, void *user
+                int fd;
+                pid_t q;
+
+                r = sd_bus_message_read(message, "h", &fd);
+                if (r < 0)
+                        return r;
+                if (r == 0)
+                        break;
+
+                r = pidfd_get_pid(fd, &q);
+                if (r < 0)
+                        return r;
+
+                r = cg_pid_get_unit(q, &unit);
+                if (r < 0)
+                        return r;
+
+                if (!streq(unit, from->id))
+                        return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS,
+                                                 "PID" PID_FMT "is not running in the same service as calling process", q);
+
+                if (!GREEDY_REALLOC(pids, n_pids+1))
+                        return -ENOMEM;
+
+                pids[n_pids++] = q;
+        }
+
+        r = sd_bus_message_exit_container(message);
+        if (r < 0)
+                return r;
+
+        r = manager_load_unit(m, name, NULL, error, &scope);
+        if (r < 0)
+                return r;
+
+        r = unit_set_slice(scope, UNIT_GET_SLICE(from));
+        if (r < 0)
+                return r;
+
+        cc = unit_get_cgroup_context(scope);
+
+        r = cgroup_context_copy(unit_get_cgroup_context(from), cc);
+        if (r < 0) {
+                log_warning_errno(r, "Failed to initialize cgroup context of the auxiliary scope, continuing with default context: %m");
+
+                cgroup_context_done(cc);
+                cgroup_context_init(cc);
+        }
+
+        r = unit_make_transient(scope);
+        if (r < 0)
+                return r;
+
+        for (unsigned i = 0; i < n_pids; i++) {
+                r = unit_pid_attachable(scope, pids[i], error);
+                if (r < 0)
+                        return r;
+
+                r = unit_watch_pid(scope, pids[i], false);
+                if (r < 0 && r != -EEXIST)
+                        return r;
+        }
+
+        /* Now load the missing bits of the unit we just created */
+        unit_add_to_load_queue(scope);
+        manager_dispatch_load_queue(m);
+
+        *u = TAKE_PTR(scope);
+
+        return 1;
+}
+
+static int method_start_aux_scope(sd_bus_message *message, void *userdata, sd_bus_error *error) {
+        Manager *m = ASSERT_PTR(userdata);
+        Unit *u;
+        int r;
+
+        assert(message);
+
+        r = mac_selinux_access_check(message, "start", error);
+        if (r < 0)
+                return r;
+
+        r = bus_verify_manage_units_async(m, message, error);
+        if (r < 0)
+                return r;
+        if (r == 0)
+                return 1; /* No authorization for now, but the async polkit stuff will call us again when it has it */
+
+        r = aux_scope_from_message(m, message, &u, error);
+        if (r < 0)
+                return r;
+
+        return bus_unit_queue_job(message, u, JOB_START, JOB_REPLACE, 0, error);
```

Comment:
The following error is false-positive, but please initialize `u` with `NULL` with short comment.
```
855s ../src/core/dbus-manager.c: In function ‘method_start_aux_scope’:
855s ../src/core/dbus-manager.c:3076:16: error: ‘u’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
855s  3076 |         return bus_unit_queue_job(message, u, JOB_START, JOB_REPLACE, 0, error);
855s       |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
855s cc1: all warnings being treated as errors
855s ninja: build stopped: subcommand failed.
```